test/issues/1497-include-error-message-on-validation-error.js

```diff
@@ -0,0 +1,53 @@
+const nock = require("nock");
+const Octokit = require("../../");
+
+require("../mocha-node-setup");
+
+describe("https://github.com/octokit/rest.js/issues/1497", () => {
+  it("octokit.repos.updateBranchProtection()", () => {
+    nock("https://request-errors-test.com", {
+      reqheaders: {
+        accept: "application/vnd.github.v3+json,application/vnd.github.hellcat-preview+json,application/vnd.github.luke-cage-preview+json,application/vnd.github.zzzax-preview+json",
+        authorization: "Bearer dXNlcm5hbWU6cGFzc3dvcmQ=",
+      },
+    })
+      .put("/repos/gr2m/sandbox/branches/gr2m-patch-1/protection", {
+        required_status_checks: { strict: true, contexts: ["wip"] },
+        enforce_admins: true,
+        required_pull_request_reviews: {
+          required_approving_review_count: 1,
+          dismiss_stale_reviews: true,
+          require_code_owner_reviews: true,
+          dismissal_restrictions: { users: [], teams: [] }
+        },
+        restrictions: { users: [], teams: [] },
+      })
+      .reply(200, {
+         message: "Validation Failed",
+         errors: [
+           "Only organization repositories can have users and team restrictions"
+         ],
+         documentation_url: "https://developer.github.com/v3/repos/branches/#update-branch-protection"
+      });
+
+    const octokit = new Octokit({
+      baseUrl: "https://request-errors-test.com",
+      auth: "Bearer dXNlcm5hbWU6cGFzc3dvcmQ=",
+    });
+    return octokit.repos.updateBranchProtection({
+      mediaType: { previews: ['hellcat', 'luke-cage', 'zzzax'] },
+      owner: "gr2m",
+      repo: "sandbox",
+      branch: "gr2m-patch-1",
+      required_status_checks: { strict: true, contexts: ["wip"] },
+      enforce_admins: true,
+      required_pull_request_reviews: {
+         required_approving_review_count: 1,
+         dismiss_stale_reviews: true,
+         require_code_owner_reviews: true,
+         dismissal_restrictions: { users: [], teams: [] }
+      },
+      restrictions: { users: [], teams: [] }
+    });
```

Comment:
You still need to check that `octokit.repos.updateBranchProtection()` rejects and that the error includes the error message. You can check that `error.message` is `Validation failed: Only organization repositories can have users and team restrictions`